README.md

```diff
@@ -200,17 +253,31 @@ If raising a Java `Exception`, throw new instance of Java `Exception` class as d
 
     throw $.newNullPointerException("Cannot copy to/from a null array.");
 
+If you need implement a native method with async JS calls, the following steps are required:
+
+1. Add the method to the `yieldMap` in jit/analyze.ts
+2. Use `asyncImpl` in override.js to return the asnyc value with a `Promise`.
+
+The `asyncImpl` is optional if part of the code doesn't make async calls, the method could return value as regular native method and the VM will handle it properly.
+
+e.g:
+
+    Native["java/lang/Thread.sleep.(J)V"] = function(delay) {
+        asyncImpl("V", new Promise(function(resolve, reject) {
+            window.setTimeout(resolve, delay.toNumber());
+        }));
+    };
```

Comment:
This example comes after the paragraph that describes how `asyncImpl` is optional, so it looks like it's supposed to be an example of that, but it's actually an example of an unconditional asynchronous native. So I would switch the order of that paragraph and the example, i.e.:

>e.g:
>```
>    Native["java/lang/Thread.sleep.(J)V"] = function(delay) {
>        asyncImpl("V", new Promise(function(resolve, reject) {
>            window.setTimeout(resolve, delay.toNumber());
>        }));
>    };
>```
>The `asyncImpl` is optional if part of the code doesn't make async calls, the method could return value as regular native method and the VM will handle it properly.

I would also rephrase and extend this description a bit, to clarify what it means for a native method to be partly asynchronous and emphasize that you always need to add such methods to `yieldMap`:

> The `asyncImpl` call is optional if part of the code doesn't make async calls. The method can sometimes return a value synchronously, and the VM will handle it properly. However, if a native ever calls `asyncImpl`, even if it doesn't always do so, then you need to add the method to `yieldMap`.